illumos-utils/src/zfs.rs

```diff
@@ -490,23 +559,27 @@ impl Zfs {
         do_format: bool,
         encryption_details: Option<EncryptionDetails>,
         size_details: Option<SizeDetails>,
+        id: Option<DatasetUuid>,
```

Comment:
I'm plumbing the UUID down so we can call `zfs set` exactly once.

Reply:
I really wanted to do this in this PR, but already felt the size of things growing a bit much. I'll file an issue and make a follow-up PR to do this.

Reply:
https://github.com/oxidecomputer/omicron/issues/7237